src/account/PluginManagerInternals.sol

```diff
@@ -886,7 +887,21 @@ abstract contract PluginManagerInternals is IPluginManager {
 
         delete _storage.pluginData[oldPlugin];
 
-        // @TODO: Apply Data migration
+        // Retrieve data from the old plugin for migration
+        bytes memory migrationData = IPlugin(oldPlugin).getDataForMigration();
+
+        // Call the old plugin's clean-up function
+        bool onReplaceOldSuccess = true;
+        try IPlugin(oldPlugin).onReplaceForOldPlugin() {}
+        catch {
+            onReplaceOldSuccess = false;
+        }
+
+        // Pass the migration data to the new plugin
+        try IPlugin(newPlugin).onReplaceForNewPlugin(migrationData) {}
+        catch (bytes memory revertReason) {
+            revert onReplaceForNewPluginFailed(newPlugin, revertReason);
+        }
 
         // @TODO: Check if onReplaceBefore succeeds and update the last parameter of the event
         emit PluginReplaced(oldPlugin, newPlugin, true);
```

Comment:
Please remove the comments above the event, and I think the `onReplaceOldSucess` should be the third parameter of the event, like the case of `onUninstall`.

Reply:
I missed that part. Thank you for the review. Will edit !